packages/client-core/src/transports/SocketWebRTCClientFunctions.ts

```diff
@@ -1157,17 +1193,43 @@ export const startScreenshare = async (network: SocketWebRTCClientNetwork) => {
   )
 
   const channelConnectionState = getState(MediaInstanceState)
+  const clientSettingState = getState(AdminClientSettingsState).client[0]
+  const screenshareSettings = clientSettingState.mediaSettings.screenshare
   const currentChannelInstanceConnection = channelConnectionState.instances[network.id]
   const channelId = currentChannelInstanceConnection.channelId
 
   await waitForTransports(network)
   const transport = MediasoupTransportState.getTransport(network.id, 'send') as WebRTCTransportExtension
 
+  let codec, encodings
+  if (screenshareSettings) {
```

Comment:
Since it's similar to the video settings we could clean this up slightly by making it a function that returns [codec, encodings].